realm/realm-library/src/androidTest/kotlin/io/realm/ManagedSetTester.kt

```diff
@@ -420,6 +421,53 @@ class ManagedSetTester<T : Any>(
         assertSetContainsSet(initializedSet.plus(notPresentValue), set)
     }
 
+    override fun putRequired() {
+        // RealmModel and Mixed setters are ignored since they cannot be marked with "@Required"
+        if (requiredSetGetter != null) {
+            val allTypesObject = createAllTypesManagedContainerAndAssert(realm)
+            assertNotNull(allTypesObject)
+            val set: RealmSet<T> = requiredSetGetter.call(allTypesObject)
+            // Check we can't insert null on a RealmSet marked as "@Required"
+            realm.executeTransaction {
+                assertFailsWith<java.lang.NullPointerException> {
+                    set.add(null)
+                }
+                assertFailsWith<java.lang.NullPointerException> {
+                    set.remove(null)
+                }
+                assertFailsWith<java.lang.NullPointerException> {
+                    set.contains(null)
+                }
+                assertFailsWith<java.lang.NullPointerException> {
+                    set.addAll(listOf(null))
+                }
+                assertFailsWith<java.lang.NullPointerException> {
+                    set.removeAll(listOf(null))
+                }
+                assertFailsWith<java.lang.NullPointerException> {
+                    set.containsAll(listOf(null))
+                }
+                assertFailsWith<java.lang.NullPointerException> {
+                    set.add(initializedSet[0])
+                    set.retainAll(listOf(null))
+                }
+            }
+
+            // Now check it works normally for the same field but without null values
```

Comment:
I guess all these are already tested by the explicit `add`, `remove`, etc. Couldn't you also just have added the above test as part of the normal tests by conditionally testing it when the field is required.